docs/DeveloperGuide.md

```diff
@@ -4,11 +4,212 @@
+
+-   ```Storage#loadSchoolClass()``` - Adds a SchoolClass to the task list without generating a successfully added message, to be used when loading from the save data.
+
+<br />
+
+**Given below is an example usage scenario for  TaskList#addSchoolClass().**
+
+**Step 1.** The user inputs a command following the proper formatting for adding a ```SchoolClass```. The Duck class will call ```Parser#processCommand()```, instantiating a ```Parser``` class, which will then call ```TaskList#addTask()```, instantiating a ```TaskList``` Class. From there, ```TaskList#addSchoolClass()``` is called and a new ```SchoolClass``` Task is instantiated, which calls ```Ui#addedTaskMessage()``` and instantiating a ```Ui``` Class. This ```SchoolClass``` Task will thus be added to the ArrayList<Task> tasks that was instantiated in the ```Duck``` class.
+
+![](https://lh4.googleusercontent.com/u4zVr8TYxFw3rMvnqdwCYlJmq0JxUgEtC_cFmOY7rPqCM9nvzcQL1t-GcTmgbedeVEHi2L6MG6xG3QaJ7XaOPs8nYvHz1Uf4wGK9bMsHDHwxZVNdS2zR79TtHL_Ub2Za0_jm6bUsnY_RQWX6QmFqCl4)
+
+![](https://lh3.googleusercontent.com/owAwOcAeTYadanERD7zj2eVD_SsbxhXUvohhaV962-DfYkgh-fV4wWVv8LnLjPpt9jl3yEkBrVUuiPg7jor-uWSpIhwdze4C3yKMWdRQEYrcR7I6tW4RMIGeyazNhEYUZzTd2BTBqUNpKL-6O2KRXug)
+  
+<br />
+
+**Given below is an example usage scenario for  Storage#loadSchoolClass.**
+
+**Step 1.** The user launches the application. The task list will be initialised with the data from a given pre-existing datafile if it exists, and the ```SchoolClass``` items will be inserted into the task list if they exist using ```Storage#loadSchoolClass()```.
+
+<br />
+
+### Recurring deadlines and events feature
+
+**Implementation**
+
+The `RecurringDeadline` and `RecurringEvent` classes are implemented to facilitate the adding of deadlines and events that happen every week to the `TaskList`. They extend `Deadline` and `Event` respectively with an additional `DayOfWeek` attribute `day` to indicate which day of the week the task recurs on. They both override the `toString()` and `toSaveString()` methods to return the correct `String` representation for output to the user and storage. `RecurringDeadline` and `RecurringEvent` also have a public method `getDay()` to return the private attribute `day`.
+
+**The following new operations are implemented. **
+
+-   `TaskList#addRecurrDeadline()` -- adds a `RecurringDeadline` object to the task list
+
+-   `Storage#loadRecurrDeadline()` -- loads a `RecurringDeadline` from the saved data file to the task list
+
+-   `TaskList#addRecurrEvent()` -- adds a `RecurringEvent` object to the task list
+
+-   `Storage#loadRecurrEvent()` -- loads a `RecurringEvent` from the saved data file to the task list
 
+Below is a class diagram for the implementation
+ 
+ <img width="859" alt="image" src="https://user-images.githubusercontent.com/88618401/228712251-dc8a6eb0-7483-4d01-b2f0-99f9c105aa9a.png">
 
+ 
+ <br />
+ 
 ## Product scope
```

Comment:
I am a bit confused about what your product is trying to achieve. I do not understand the functionality of the product is it a deadline tracker or what as there is no specification on what the product helps the user in. Could also add a few diagrams for the rest of the features.